test/fake-test.js

```diff
@@ -0,0 +1,193 @@
+"use strict";
+
+var sinon = require("../lib/sinon.js");
+var fake = sinon.fake;
+var referee = require("referee");
+var assert = referee.assert;
+
+referee.add("isProxy", {
+    assert: function assertIsProxy(actual) {
+        if (typeof actual !== "function") {
+            throw new TypeError("isProxy expects 'actual' argument to be a Function");
+        }
+
+        this.isSinonProxy = actual.isSinonProxy;
+
+        return this.isSinonProxy === true;
+    },
+    assertMessage: "Expected ${name} to be a Sinon proxy (spy)",
+    refuteMessage: "Expected ${name} to not be a Sinon proxy (spy)",
+    expectation: "toBeAProxy"
+});
+
+function verifyProxy(func, argument) {
+    it("should return a Sinon proxy", function () {
+        var actual = func(argument);
+
+        assert.isProxy(actual);
+    });
+}
+
+function noop() {}
+
+var hasFunctionNameSupport = typeof noop.name !== "undefined";
```

Comment:
`noop.name === 'noop'` is a bit more straight forward.

Reply:
Good point, thanks :)